ironfish/src/wallet/walletdb/walletdb.ts

```diff
@@ -572,4 +584,38 @@ export class WalletDB {
   ): Promise<void> {
     await this.pendingTransactionHashes.clear(tx, account.prefixRange)
   }
+
+  async cleanupDeletedAccounts(signal?: AbortSignal): Promise<void> {
+    let recordsToCleanup = 1000
+
+    const stores: IDatabaseStore<{
+      key: Readonly<unknown>
+      value: unknown
+    }>[] = [
+      this.transactions,
+      this.sequenceToNoteHash,
+      this.nonChainNoteHashes,
+      this.nullifierToNoteHash,
+      this.pendingTransactionHashes,
+      this.decryptedNotes,
+    ]
+
+    for (const [accountId] of await this.accountIdsToCleanup.getAll()) {
+      const prefix = calculateAccountPrefix(accountId)
+      const range = StorageUtils.getPrefixKeyRange(prefix)
+
+      for (const store of stores) {
+        for await (const key of store.getAllKeysIter(undefined, range)) {
+          if (signal?.aborted === true || recordsToCleanup === 0) {
+            return
+          }
+
+          await store.del(key)
```

Comment:
does this need try catch in case the del is not successful?

Reply:
What would you expect the catch to do? If If the delete is not successful the program should crash because the leveldb is corrupt and can no longer operate.

Reply:
I just looked up, `del` does not throw any error.